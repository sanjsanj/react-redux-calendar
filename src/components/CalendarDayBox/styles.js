import styled from "styled-components";

export const StyledCalendarDayBox = styled.div`
  background: ${props =>
    props.disabled
      ? props.theme.disabledDayBackgroundColor
      : props.theme.dayBackgroundLinearGradient};

  cursor: ${props => (props.disabled ? "auto" : "pointer")};
  color: ${props => props.theme.mainTextColor};
  transition: all 0.3s ease-in-out;
  overflow: hidden;
  margin: 1px auto;
  padding: 5px;
  height: 80px;

  :hover {
    box-shadow: ${props => (props.disabled ? "" : "0px 3px 9px #000000")};
  }

  ol {
    list-style: none;
    padding: 0;
    margin: 0;
  }

  p {
    margin: 0 auto 5px;
  }

  li {
    margin: 0 auto 5px;
    font-size: 11px;
  }
`;
