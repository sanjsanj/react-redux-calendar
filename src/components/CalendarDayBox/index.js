import React from "react";
import PropTypes from "prop-types";

import { StyledCalendarDayBox } from "./styles";

export default function CalendarDayBox(props) {
  return (
    <StyledCalendarDayBox disabled={!props.day}>
      {props.day && props.day.dayOfMonth}
    </StyledCalendarDayBox>
  );
}

CalendarDayBox.propTypes = {
  day: PropTypes.shape({ dayOfMonth: PropTypes.number.isRequired }),
};
