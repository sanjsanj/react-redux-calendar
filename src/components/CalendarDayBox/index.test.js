import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import CalendarDayBox from ".";

describe("CalendarDayBox", () => {
  describe("when passed no day prop", () => {
    const wrapper = shallow(<CalendarDayBox day={undefined} />);

    it("should not have any content", () => {
      expect(wrapper.text()).toEqual("");
    });

    it("should match snapshot", () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe("when passed a day prop", () => {
    const wrapper = shallow(
      <CalendarDayBox day={{ dayOfMonth: 28, reminders: [] }} />
    );

    it("should have day of month", () => {
      expect(wrapper.text()).toEqual("28");
    });

    it("should show reminders for that day", () => {});

    it("should match snapshot", () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
