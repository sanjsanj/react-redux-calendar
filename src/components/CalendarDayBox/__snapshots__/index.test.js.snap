// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDayBox when passed a day prop should match snapshot 1`] = `
<styled.div
  disabled={false}
>
  28
</styled.div>
`;

exports[`CalendarDayBox when passed no day prop should match snapshot 1`] = `
<styled.div
  disabled={true}
/>
`;
